File: templates/griffon-standard-templates/templates/griffon-swing-java/griffon-app/controllers/Controller.java
package ${project_package};

import griffon.core.GriffonApplication;
import griffon.core.artifact.GriffonController;
import griffon.metadata.ArtifactProviderFor;
import org.codehaus.griffon.runtime.core.artifact.AbstractGriffonController;

import javax.annotation.Nonnull;
import javax.inject.Inject;
import griffon.transform.Threading;

@ArtifactProviderFor(GriffonController.class)
public class ${project_capitalized_name}Controller extends AbstractGriffonController {
    private ${project_capitalized_name}Model model;

    @Inject
    public ${project_capitalized_name}Controller(@Nonnull GriffonApplication application) {
        super(application);
    }

    public void setModel(${project_capitalized_name}Model model) {
        this.model = model;
    }

    @Threading(Threading.Policy.INSIDE_UITHREAD_ASYNC)
    public void click() {
        model.setClickCount(model.getClickCount() + 1);
    }
}

File: templates/griffon-javafx-kotlin-templates/templates/subtmpl-artifact/Controller.kt
package ${project_package}

import griffon.core.artifact.GriffonController
import griffon.core.controller.ControllerAction
import griffon.inject.MVCMember
import griffon.metadata.ArtifactProviderFor
import org.codehaus.griffon.runtime.core.artifact.AbstractGriffonController
import javax.annotation.Nonnull
import javax.application.threading.Threading

@ArtifactProviderFor(GriffonController::class)
class ${project_class_name}Controller : AbstractGriffonController() {
    @set:[MVCMember Nonnull]
    lateinit var model: ${project_class_name}Model

    @ControllerAction
    @Threading(Threading.Policy.INSIDE_UITHREAD_ASYNC)
    fun click() {
        val count = Integer.parseInt(model.clickCount)
        model.clickCount = (count + 1).toString()
    }
}